tests/generic_config_updater/gu_utils.py

```diff
@@ -305,18 +305,21 @@ def get_asic_name(duthost):
     asic_mapping = gcu_conf["helper_data"]["rdma_config_update_validator"]
     if asic_type == 'cisco-8000':
         asic = "cisco-8000"
-    elif asic_type == 'mellanox' or asic_type == 'vs' or asic_type == 'broadcom':
-        hwsku = duthost.shell(GET_HWSKU_CMD)['stdout'].rstrip('\n')
-        if asic_type == 'mellanox' or asic_type == 'vs':
-            spc1_hwskus = asic_mapping["mellanox_asics"]["spc1"]
-            if hwsku.lower() in [spc1_hwsku.lower() for spc1_hwsku in spc1_hwskus]:
-                asic = "spc1"
-                return asic
-        if asic_type == 'broadcom' or asic_type == 'vs':
+    elif asic_type == 'vs':
+        asic = "vs"
+    elif asic_type == 'mellanox' or asic_type == 'broadcom':
+        cur_hwsku = duthost.shell(GET_HWSKU_CMD)['stdout'].rstrip('\n')
+        if asic_type == 'mellanox':
+            asic_hwskus = asic_mapping["mellanox_asics"]
+            for asic_name, hwskus in asic_hwskus.items():
+                if cur_hwsku.lower() in [hwsku.lower() for hwsku in hwskus]:
```

Comment:
Since the logic is the same between mellanox and broadcom now, could we make the variable naming convention and loop logic consistent between the two?

Reply:
Fixed.